content/en/platform/corda/5.0-beta/developing/development-tutorials/build-basic-cordapp/basic-cordapp-flows.md

```diff
@@ -0,0 +1,796 @@
+
+    @Suspendable
+    override fun call(requestBody: RPCRequestData): String {
+
+    }
+}
+```
+
+
+#### Extract the `CreateAndIssueAppleStampFlow`'s Request Parameters
+
+`RPCRequestData` contains the flow’s request parameters passed in via HTTP. To extract the data from the RPCRequestData, you should create a class that represents the data.
+
+1. Create `CreateAndIssueAppleStampRequest` with the following properties:
+
+* `stampDescription` - A `String` description for the `AppleStamp`.
+* `holder` - A `MemberX500Name` for the participant who is issued an `AppleStamp`.
+
+This is how it should look like:
+
+```kotlin
+data class CreateAndIssueAppleStampRequest(
+    val stampDescription: String,
+    val holder: MemberX500Name
+)
+```
+
+2. To extract the request data into a `CreateAndIssueAppleStampRequest`, add `RPCRequestData.getRequestBodyAs` to `CreateAndIssueAppleStampFlow`.
+
+3. Add variables for `stampDescription` and `holderName` and extract them from the `CreateAndIssueAppleStampRequest` instance.
+
+Your code should now look like this:
+
+```kotlin
+@InitiatingFlow(protocol = "create-and-issue-apple-stamp")
+class CreateAndIssueAppleStampFlow : RPCStartableFlow {
+
+    @CordaInject
+    lateinit var flowMessaging: FlowMessaging
+
+    @CordaInject
+    lateinit var jsonMarshallingService: JsonMarshallingService
+
+    @CordaInject
+    lateinit var memberLookup: MemberLookup
+
+    @CordaInject
+    lateinit var notaryLookup: NotaryLookup
+
+    @CordaInject
+    lateinit var utxoLedgerService: UtxoLedgerService
+
+    @Suspendable
+    override fun call(requestBody: RPCRequestData): String {
+        val request = requestBody.getRequestBodyAs<CreateAndIssueAppleStampRequest>(jsonMarshallingService)
+        val stampDescription = request.stampDescription
+        val holderName = request.holder
+    }
+}
+```
+
+
+#### Obtain a Reference for the Notary
+
+In transactions with multiple parties, you need a notary service to reach consensus between the parties. Since this flow is communicating with Farmer Bob and Peter, you need the notary.
+
+Add the following code to retrieve a notary:
+
+```kotlin
+// Retrieve the notaries public key (this will change)
+val notaryInfo = notaryLookup.notaryServices.single()
+val notaryKey = memberLookup.lookup().single {
+    it.memberProvidedContext["corda.notary.service.name"] == notaryInfo.name.toString()
+}.ledgerKeys.first()
+val notary = Party(notaryInfo.name, notaryKey)
+```
+
+
+#### Lookup the Issuer and Holder of the New `AppleStamp` State
+
+1. Use `memberLookup.myInfo` to lookup the current participant (who is executing the flow).
+
+2. Extract the `name` and `ledgerKey`s (take the first one) of the participant and store the result in a `Party`.
+
+3. Repeat the same process for the holder of the new `AppleStamp` state using the `holderName` from the flow’s input parameters.
+
+4. Verify that this participant exists in the network.
```

Comment:
@lankydan Which participant? The one that is executing the flow or holder or both?

Reply:
The participant with the passed in `holderName`.

Reply:
Updated.